File: 241008/k개 중에 1개를 n번 뽑기/n-permutations-of-k-with-repetition.py
import sys

def choose():
    global answer
    
    # 종료 조건
    if len(answer) == n: 
        print(*answer)
        return
    
    # 재귀 조건
    for i in range(1, k + 1):
        answer.append(i)
        choose()  # Recursive call with next number
        answer.pop()  # Backtrack to try other numbers

k, n = map(int, sys.stdin.readline().split())
answer = []
choose()

File: 241008/k개 중에 1개를 n번 뽑기/n-permutations-of-k-with-repetition.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> answer;
int K, N;

void choose(){
    if(answer.size() == 2){
        for(auto a: answer){
            cout << a << " ";
        }
        cout << "\n";
    }

    else{
        for(int i=1; i <= K; i++){
            answer.emplace_back(i);
            choose();
            answer.pop_back();

        }
    }
}

int main() {
    cin >> K >> N;
    choose();
    return 0;
}